packages/react/src/components/TopNav/TopNav.tsx

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, {FC, ReactElement, useState} from 'react';
+import clsx from 'clsx';
+import MuiAppBar, {AppBarProps as MuiAppBarProps} from '@mui/material/AppBar';
+import {Box, Container, IconButton, Toolbar, Typography, useMediaQuery} from '@mui/material';
+import {
+  ChevronDownIcon,
+  EclipseIcon,
+  SunIcon,
+  HalfStripedSunIcon,
+  MoonWithRaysIcon,
+  HamburgerIcon,
+} from '@oxygen-ui/react-icons';
+import {useTheme, Theme} from '@mui/material/styles';
+import {WithWrapperProps} from '../../models';
+import {composeComponentDisplayName} from '../../utils';
+import './top-nav.scss';
+import Avatar from '../Avatar';
+import Button from '../Button';
+import Link from '../Link';
+import UserDropdownMenu, {ThemeListInterface} from '../UserDropdownMenu';
+
+export interface TopNavProps extends MuiAppBarProps {
+  /**
+   * Handle left navigation bar button toggle.
+   */
+  handleLeftNav?: () => void;
+  /**
+   * Log out function.
+   */
+  handleLogOut?: () => void;
+  /**
+   * Path to navigate to the Home/Landing page.
+   */
+  homePath?: string;
```

Comment:
Change this to a callback.

Reason:
Home path is not always an absolute URL. It could be a relative path. We can't handle that inside here.

```suggestion
  onBrandClick?: string;
```

Reply:
Fixed in latest commits